### PR TITLE
Auth setup

### PR DESCRIPTION
**Pull request by NacymeG**

### CODE REVIEW

File: db-config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require('dotenv').config();
+const mysql = require('mysql2');
+
+const connection = mysql.createConnection({
```
Comment (Caroline-CD) [inline db-config.js:6]: It is possible to lighten the code for to avoid much process.env no ?

File: models/auth.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+const signupValidation = (data, forCreation = true) => {
+  const presence = forCreation ? 'required' : 'optional';
+  return Joi.object({
+    mail: Joi.string().email().max(128).presence(presence),
+    password: Joi.string()
+      .max(64)
+      .presence(presence)
+      .pattern(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/),
+    repeadPass: Joi.ref('password'),
```
Comment (Caroline-CD) [inline models/auth.js]: There is a little mistake here : repeatPass and no repeadPass

File: routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    res.status(404).send(`Identifiant ou mot de passe incorrect
+  `);
+  const checkPwd = await Auth.verifyPassword(password, user.password);
+  if (checkPwd && user) {
+    const jwt = await Auth.calculateToken(mail, user.id);
+    res
+      .status(202)
+      .cookie('user_token', jwt)
+      .send(`Hi ${user.firstname} ${user.lastname} !`);
+    console.log('Cookies: ', req.headers.cookie.split('=')[1]);
```
Comment (Caroline-CD) [inline routes/auth.js]: It will be necessary to remove the console.log

**Review by Caroline-CD — APPROVED** [review]
Some small modifications.

**Comment by LoicBrassart** [comment]
@Caroline-CD If you spot changes to do, don't approve the PR ;-)

**Review by Caroline-CD — CHANGES_REQUESTED** [review]
Sorry, there are small modifications to be made

**Review by Caroline-CD — APPROVED** [review]
Ok :crystal_ball:

**Review by Hacene010 — APPROVED** [review]
it seems ok for me but why didn't you use passport ?

**Comment by NacymeG** [comment]
> it seems ok for me but why didn't you use passport ?

The next PR will focus on the refactoring of authentication and passport implementation.

**Pull request merged**